Terminal.Gui/ConsoleDrivers/WindowsDriver.cs

```diff
@@ -32,44 +33,131 @@ internal class WindowsConsole {
 	CursorVisibility? _currentCursorVisibility = null;
 	CursorVisibility? _pendingCursorVisibility = null;
 	readonly StringBuilder _stringBuilder = new StringBuilder (256 * 1024);
+	bool _isWindowsTerminal;
 
-	public WindowsConsole ()
+	public WindowsConsole (bool isWindowsTerminal)
 	{
 		_inputHandle = GetStdHandle (STD_INPUT_HANDLE);
 		_outputHandle = GetStdHandle (STD_OUTPUT_HANDLE);
-		_originalConsoleMode = ConsoleMode;
-		var newConsoleMode = _originalConsoleMode;
-		newConsoleMode |= (uint)(ConsoleModes.EnableMouseInput | ConsoleModes.EnableExtendedFlags);
-		newConsoleMode &= ~(uint)ConsoleModes.EnableQuickEditMode;
-		newConsoleMode &= ~(uint)ConsoleModes.EnableProcessedInput;
-		ConsoleMode = newConsoleMode;
+
+		_isWindowsTerminal = isWindowsTerminal;
+		if (!ConsoleDriver.RunningUnitTests && !_isWindowsTerminal) {
+			_originalScreenBufferInfo = new CONSOLE_SCREEN_BUFFER_INFOEX ();
+			_originalScreenBufferInfo.cbSize = (uint)Marshal.SizeOf (_originalScreenBufferInfo);
+			if (!GetConsoleScreenBufferInfoEx (_outputHandle, ref _originalScreenBufferInfo)) {
+				var err = Marshal.GetLastWin32Error ();
+				if (err != 0) {
+					throw new System.ComponentModel.Win32Exception (err);
+				}
+			}
+		}
+
+		_originalInputConsoleMode = InputConsoleMode;
+		var newInConsoleMode = _originalInputConsoleMode;
+		newInConsoleMode |= (uint)(ConsoleModes.EnableMouseInput | ConsoleModes.EnableExtendedFlags);
+		newInConsoleMode &= ~(uint)ConsoleModes.EnableQuickEditMode;
+		newInConsoleMode &= ~(uint)ConsoleModes.EnableProcessedInput;
+		InputConsoleMode = newInConsoleMode;
+
+		_originalOutputConsoleMode = OutputConsoleMode;
+		var newOutConsoleMode = _originalOutputConsoleMode;
+		if (!ConsoleDriver.RunningUnitTests && (newOutConsoleMode & (uint)(ConsoleModes.EnableVirtualTerminalProcessing | ConsoleModes.DisableNewlineAutoReturn))
+			< (uint)ConsoleModes.DisableNewlineAutoReturn) {
+
+			newOutConsoleMode |= (uint)(ConsoleModes.EnableVirtualTerminalProcessing | ConsoleModes.DisableNewlineAutoReturn);
+			OutputConsoleMode = newOutConsoleMode;
+		}
 	}
 
 	CharInfo [] _originalStdOutChars;
 
-	public bool WriteToConsole (Size size, ExtendedCharInfo [] charInfoBuffer, Coord bufferSize, SmallRect window, bool force16Colors)
+	public bool WriteToConsole (Size size, ExtendedRuneInfo [,] charInfoBuffer, Coord bufferSize, SmallRect window, bool force16Colors)
 	{
 		if (_screenBuffer == IntPtr.Zero) {
 			ReadFromConsoleOutput (size, bufferSize, ref window);
 		}
 
 		bool result = false;
+
 		if (force16Colors) {
-			int i = 0;
-			CharInfo [] ci = new CharInfo [charInfoBuffer.Length];
-			foreach (ExtendedCharInfo info in charInfoBuffer) {
-				ci [i++] = new CharInfo () {
-					Char = new CharUnion () { UnicodeChar = info.Char },
-					Attributes = (ushort)(((int)info.Attribute.Foreground.ColorName) | ((int)info.Attribute.Background.ColorName << 4))
-				};
+			var csbi = new CONSOLE_SCREEN_BUFFER_INFOEX ();
```

Comment:
As far as I can tell, this force16 color stuff has nothing to do with the Chinese issue. 

Can it be removed from this PR? Perhaps we can fold this into the work I'm doing here: https://github.com/gui-cs/Terminal.Gui/pull/2940

Reply:
@tig with this change I aldo resolved the non-bmp displaying on `WindowsDriver` without using `Windows Terminal`.

![image](https://github.com/gui-cs/Terminal.Gui/assets/13117724/e8c56bfc-8a03-4877-b180-a19417bfcd96)